proposals/text/5289-recentness-ranker.md

```diff
@@ -0,0 +1,164 @@
+    top_k: 3
+    method: score
+- name: qa_template
+  type: PromptTemplate
+  params:
+    output_parser:
+        type: AnswerParser
+    prompt: "prompt text here"
+- name: PromptNode
+  type: PromptNode
+  params:
+    default_prompt_template: qa_template
+    max_length: 300
+    model_kwargs:
+      temperature: 0
+    model_name_or_path: gpt-3.5-turbo
+- name: FileTypeClassifier
+  type: FileTypeClassifier
+- name: TextConverter
+  type: TextConverter
+- name: PDFConverter
+  type: PDFToTextConverter
+- name: Preprocessor
+  params:
+    language: en
+    split_by: word
+    split_length: 200
+    split_overlap: 10
+    split_respect_sentence_boundary: true
+  type: PreProcessor
+
+pipelines:
+- name: query
+  nodes:
+    - name: EmbeddingRetriever
+      inputs: [Query]
+    - name: BM25Retriever
+      inputs: [Query]
+    - name: JoinDocuments
+      inputs: [EmbeddingRetriever, BM25Retriever]
+    - name: Ranker
+      inputs: [JoinDocuments]
+    - name: RecentnessRanker
+      inputs: [Ranker]
+    - name: PromptNode
+      inputs: [RecentnessRanker]
+
+- name: indexing
+  nodes:
+  - inputs:
+    - File
+    name: FileTypeClassifier
+  - inputs:
+    - FileTypeClassifier.output_1
+    name: TextConverter
+  - inputs:
+    - FileTypeClassifier.output_2
+    name: PDFConverter
+  - inputs:
+    - TextConverter
+    - PDFConverter
+    name: Preprocessor
+  - inputs:
+    - Preprocessor
+    name: EmbeddingRetriever
+  - inputs:
+    - EmbeddingRetriever
+    name: DocumentStore
+
+  ```
+
+# Motivation
+
+Initially this ranker was implemented by Timo for a customer case where the date of the document mattered for retrieval. The reason we would like to add it to Haystack is because we see wider use for this node in future customer and community cases. One example where document recency matters is in a QA solution based on technical documentation with release notes of a software product - the older release notes should naturally have less priority in the responses than the most recent ones. And another example is news content - news articles retrieval can definitely benefit from recency being factored into the relevance calculation.
+
+# Detailed design
+
+The ranker has already been implemented by Timo here: https://github.com/deepset-ai/deepset-cloud-custom-nodes/blob/main/deepset_cloud_custom_nodes/rankers/recentness_reranker.py & PR was reviewed at the time by Seb and Florian: https://github.com/deepset-ai/deepset-cloud-custom-nodes/pull/54.
+
+Additionally, you can see the code for this proposal here: https://github.com/deepset-ai/haystack/pull/5301/files. It is the same code as above, just with small naming changes (e.g. "rff" method got changed to "reciprocal_rank_fusion" to match the existing naming used Haystack's JoinDocuments node)
+
+As a general description, the ranker works by sorting the documents based on date and modifying the relevance score calculation slightly to include recency-based weight.
```

Comment:
As the custom nodes repository is private, most of our community doesn't have access to it. Therefore, it would be nice if you could briefly describe the design here. Especially, it would be nice to answer the following questions:
- How are the Documents scored based on the provided date?
- How is the date score combined with the retriever score?
- What methods do we support as values for the `method` parameter?

Reply:
I have expanded the "Detailed design" section - feel free to take a look & let me know if it can be further improved 👍